src/H5Odeprec.c

```diff
@@ -127,6 +129,19 @@ H5O__iterate1_adapter(hid_t obj_id, const char *name, const H5O_info2_t *oinfo2,
     assert(oinfo2);
     assert(op_data);
 
+    /* Get the VOL object */
+    if (NULL == (vol_obj = H5VL_vol_object(obj_id)))
+        HGOTO_ERROR(H5E_ARGS, H5E_BADTYPE, FAIL, "invalid location identifier");
+
+    /* Check if using native VOL connector */
+    if (H5VL_object_is_native(vol_obj, &is_native) < 0)
```

Comment:
This is probably fine, but I think it would be better to pt this check at the top level, in HOvisit1(), etc. Even though it needs to be in mroe places I think it's better to catch it before it goes into the VOL connector at all and give a cleaner error message.